L1Trigger/L1THGCal/test/HGCalTriggerGeomTesterV9Imp2.cc

```diff
@@ -0,0 +1,1155 @@
+            moduleTC_z_     .get()[itc] = position.z();
+            itc++;
+        }
+        auto cells_in_module = triggerGeometry_->getCellsFromModule(id);
+        moduleCell_N_   = cells_in_module.size();
+        //
+        setTreeModuleCellSize(moduleCell_N_);
+        size_t ic = 0;
+        for(const auto& c : cells_in_module)
+        {
+            if(id.det()==DetId::HGCalHSc)
+            {
+                HGCScintillatorDetId cId(c);
+                GlobalPoint cell_position = triggerGeometry_->hscGeometry()->getPosition(cId);
+                triggerCellCell_id_    .get()[ic] = c;
+                triggerCellCell_zside_ .get()[ic] = cId.zside();
+                triggerCellCell_subdet_.get()[ic] = cId.subdetId();
+                triggerCellCell_layer_ .get()[ic] = cId.layer();
+                triggerCellCell_waferU_ .get()[ic] = 0;
+                triggerCellCell_waferV_ .get()[ic] = 0;
+                triggerCellCell_cellU_  .get()[ic] = 0;
+                triggerCellCell_cellV_  .get()[ic] = 0;
+                triggerCellCell_ieta_  .get()[ic] = cId.ietaAbs();
+                triggerCellCell_iphi_  .get()[ic] = cId.iphi();
+                triggerCellCell_x_     .get()[ic] = cell_position.x();
+                triggerCellCell_y_     .get()[ic] = cell_position.y();
+                triggerCellCell_z_     .get()[ic] = cell_position.z();
+            }
+            else
+            {
+                HGCSiliconDetId cId(c);
+                const GlobalPoint position = (cId.det()==DetId::HGCalEE ? triggerGeometry_->eeGeometry()->getPosition(cId) :  triggerGeometry_->hsiGeometry()->getPosition(cId));
+                moduleCell_id_    .get()[ic] = c;
+                moduleCell_zside_ .get()[ic] = cId.zside();
+                moduleCell_subdet_.get()[ic] = cId.subdetId();
+                moduleCell_layer_ .get()[ic] = cId.layer();
+                moduleCell_waferU_ .get()[ic] = cId.waferU();
+                moduleCell_waferV_ .get()[ic] = cId.waferV();
+                moduleCell_cellU_  .get()[ic] = cId.cellU();
+                moduleCell_cellV_  .get()[ic] = cId.cellV();
+                moduleCell_x_     .get()[ic] = position.x();
+                moduleCell_y_     .get()[ic] = position.y();
+                moduleCell_z_     .get()[ic] = position.z();
+                ic++;
+            }
+        }
+        //
+        treeModules_->Fill();
+    }
+
+}
+
+
+/*****************************************************************/
+void HGCalTriggerGeomTesterV9Imp2::analyze(const edm::Event& e, 
+        const edm::EventSetup& es) 
+/*****************************************************************/
+{
+
+}
+
+
+/*****************************************************************/
+void HGCalTriggerGeomTesterV9Imp2::setTreeModuleSize(const size_t n) 
+/*****************************************************************/
+{
+    moduleTC_id_    .reset(new int[n],   array_deleter<int>());
```

Comment:
why not just use a vector and set the branch address to `vector::data()`?

Reply:
No real reason. This is some old code which has been copy-pasted several times.
I can switch to vectors in the future. The same thing is done in several tester classes, it's not just in this one.